docs/api/hashlib.rst

```diff
@@ -0,0 +1,83 @@
+nacl.hashlib
+============
+
+The :py:mod:`nacl.hashlib` module exposes directly usable implementations
+of raw constructs which libsodium exposes with simplified APIs, like the
+ones in :py:mod:`nacl.hash` and in :py:mod:`nacl.pw_hash`.
+
+When such constructs are exposed in some cpython version standard library,
```

Comment:
I don't think this paragraph is what we want to say. The current implementations are compatible with the standard library is all we really need to say. The stdlib is extraordinarily unlikely to break backwards compatibility